src/audio/buffer.c

```diff
@@ -214,6 +257,63 @@ int buffer_set_size(struct comp_buffer *buffer, uint32_t size, uint32_t alignmen
 	return 0;
 }
 
+int buffer_set_size_range(struct comp_buffer *buffer, size_t preferred_size, size_t minimum_size,
+			  uint32_t alignment)
+{
+	void *ptr = audio_stream_get_addr(&buffer->stream);
+	const size_t actual_size = audio_stream_get_size(&buffer->stream);
+	void *new_ptr = NULL;
+	size_t new_size;
+
+	CORE_CHECK_STRUCT(buffer);
+
+	/* validate request */
+	if (minimum_size == 0 || preferred_size < minimum_size) {
+		buf_err(buffer, "resize size range %zu -- %zu is invalid", minimum_size,
+			preferred_size);
+		return -EINVAL;
+	}
+
+	/* Align preferred size to a multiple of the minimum size */
+	if (preferred_size % minimum_size)
+		preferred_size += minimum_size - preferred_size % minimum_size;
```

Comment:
in fact wondering, whether it would be better to align down? Doesn't seem important though